tests/test_materialized_view_helpers.py

```diff
@@ -0,0 +1,142 @@
+
+Base = declarative_base()
+metadata = Base.metadata
+
+
+class Thing(Base):
+    __tablename__ = 'things'
+    id = Column(Integer, primary_key=True)
+    name = Column(String)
+    description_id = Column(Integer, ForeignKey('descriptions.id'))
+
+    def __repr__(self):
+        return '<Thing(id=%d, name=%s, desc_id=%d)>' % (self.id, self.name, self.description_id)
+
+
+class Description(Base):
+    __tablename__ = 'descriptions'
+    id = Column(Integer, primary_key=True)
+    desc = Column(String)
+
+    def __repr__(self):
+        return '<Description(id=%d, desc=%s)>' % (self.id, self.desc)
+
+
+class SimpleThing(Base, MaterializedViewMixin):
+    # __selectable__ = select([Thing]).where(Thing.id <= literal_column('3'))  # deeelightful
+
+    # less delightful but applicable to cases where we use text queries for selectable:
+    __selectable__ = text("""
+        SELECT *
+        FROM things
+        WHERE things.id <= 3
+    """).columns(Thing.id, Thing.name, Thing.description_id)
+
+    def __repr__(self):
+        return '<SimpleThing(id=%d, desc=%s)>' % (self.id, self.name)
+
+
+class ThingWithDescription(Base, MaterializedViewMixin):
+    # this works:
+    # __selectable__ = select([Thing.id, Thing.name, Description.desc])\
+    #     .select_from(Thing.__table__.join(Description.__table__, Thing.description_id == Description.id))
+
+    # this works:
+    # __selectable__ = text('''
+    #     SELECT things.id, things.name, descriptions.desc
+    #     FROM things JOIN descriptions ON (things.description_id = descriptions.id)
+    # ''').columns(Thing.id, Thing.name, Description.desc)
+
+    # so does this! which is essential for text-defined queries with new columns
+    __selectable__ = text('''
+        SELECT things.id, things.name, descriptions.desc
+        FROM things JOIN descriptions ON (things.description_id = descriptions.id)
+    ''').columns(column('id'), column('name'), column('desc'))
+    __primary_key__ = ['id']
+
+
+class ThingCount(Base, MaterializedViewMixin):
+    __selectable__ = text("""
+        SELECT d.desc as desc, count(things.id) as num
+        FROM things JOIN descriptions as d ON (things.description_id = d.id)
+        GROUP BY d.desc
+    """).columns(column('desc'), column('num'))
+    __primary_key__ = ['desc']
+
+    def __repr__(self):
+        return '<ThingWithDescription(id=%d, name=%s, desc=%s)>' % (self.id, self.name, self.desc)
+
+
+@fixture(scope="module")
+def mod_empty_test_db_session(mod_blank_postgis_session):
+    # logging.getLogger('sqlalchemy.engine').setLevel(logging.INFO)
+    sesh = mod_blank_postgis_session
+    engine = sesh.get_bind()
+    Base.metadata.create_all(bind=engine)
+    views = [SimpleThing, ThingWithDescription, ThingCount]
+    for view in views:
+        view.create(sesh)
+    yield sesh
+    for view in reversed(views):
+        view.drop(sesh)
+
+@fixture
+def view_test_session(mod_empty_test_db_session):
```

Comment:
The test functions that you eventually write show the elegance of this approach. Everything comes together with a very simple and easy-to-read usage. I like it.